modules/programs/granted.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.granted;
+  package = pkgs.granted;
+
+in {
+  meta.maintainers = [ hm.maintainers.wcarlsen ];
+
+  options.programs.granted = {
+    enable = mkEnableOption "granted";
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+  };
+
+  config = let
+    manualShellProfile = ''
+      alias assume="source ${package}/bin/.assume-wrapped"
```

Comment:
This honestly seems like a packaging bug to me. If there is no hope to run the assume script as an actual script then it shouldn't be in `/bin` at all, instead something like `/share/bash/assume.sh` would be more suitable. The file could be patched to unconditionally set (and unset) the `GRANTED_ALIAS_CONFIGURED` variable so the user doesn't have to worry about that.

Perhaps could place a script in `/bin/assume` that contains something like

``` sh
cat > /dev/stderr <<EOF
It is not possible to run 'assume' directly, you must create a shell alias.

Bash:  alias assume='{nix profile path}/share/bash/assume.sh'
Zsh:    alias assume='{nix profile path}/share/bash/assume.sh'
Fish:    …
EOF

exit 1
```

When the user has created an alias, then I think all shells would run the alias over the above assume script.

Reply:
Packaging is not ideal from granted and I totally agree about the `/bin` part. I think the alias with source comes into play since because otherwise the cannot set environment variables for an AWS session in you current shell. Unfortunately I'm not skilled enough in Nix yet to do advanced stuff in packaging.

Reply:
@rycee I have hard time resolving this one, since it seems to be on the packaging side of things. Anything I can do to progress?

Reply:
Yeah, my comment here is only for the package in Nixpkgs. Not relevant for this module (except that the module needs to be updated in case the package in Nixpkgs is updated in the way I suggest above).